CMakeLists.txt

```diff
@@ -269,22 +269,25 @@ set_source_files_properties(third_party/jsoncpp/jsoncpp.cpp PROPERTIES COMPILE_F
 add_library(jsoncpp OBJECT third_party/jsoncpp/jsoncpp.cpp)
 
 if (CMAKE_COMPILER_IS_GNUCXX)
-    add_definitions(-fstack-protector-all)
+    # Enable run-time overflow checks. We are not performance-critical enough
+    # for this to be too expensive.
+    add_compile_options(-fstack-protector-all)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=undefined -fsanitize=bounds")
     # Enable maximum set of warnings. -Wno-sign-compare is required because of
     # problems in gtest. -Wswitch-default and -Wconversion would be nice as
     # well, but they also cause problems in gtest.
-    add_definitions(-Wall -Wextra -Wformat-security -Wfloat-equal -Wcast-qual -Wlogical-op -Wno-sign-compare)
+    add_compile_options(-Wall -Wextra -Wformat-security -Wfloat-equal -Wcast-qual -Wlogical-op -Wno-sign-compare)
 
     if (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "4.9" OR MAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.9")
```

Comment:
Unrelated to your changes, but it looks like there is a typo there, it was probably intended to be `CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "4.9"`.

It generates a fair amount of warning when fixed!

Reply:
Wow, I made that type in what must have been my second week here. Oops. I vote for fixing that in a separate PR. Do you want to look into that?

Reply:
Sure :). I guess you can blame CMake weirdness and general shell script style programming.

Reply:
@patrickvacek #782